Source/Classes/BuildCompleter.ps1

```diff
@@ -0,0 +1,16 @@
+class BuildCompleter : System.Management.Automation.IArgumentCompleter {
+  [System.Collections.Generic.IEnumerable[System.Management.Automation.CompletionResult]] CompleteArgument(
+        [string]$CommandName, [string]$ParameterName, [string]$WordToComplete,
+        [System.Management.Automation.Language.CommandAst]$CommandAst, [System.Collections.IDictionary] $FakeBoundParameters
+  )    {
+        $results = [System.Collections.Generic.List[System.Management.Automation.CompletionResult]]::new()
+        if ($Global:PSDefaultParameterValues["*:projectName"]){
+            foreach ($b in (Get-VSTeamBuild -ProjectName $Global:PSDefaultParameterValues["*:projectName"]).name ) {
```

Comment:
It might be to do with strictmode . When it's off  ($null).name is fine,   , when it is on it gives  that error. I noticed the tests turn strictmode on, and I have a tendency to write code which relies on it being off.  If you need/want to change that anywhere feel free to do so :-) 

There are a couple of cases where dynamic parameters may still be the most sensible way to do it - that's usually where the value of parameter A determines if parameter B is visible or not. And I may have left some in that are only used once.

Reply:
Got it! Good to know.